src/chains/tests/test_signals_ff_hook_events.py

```diff
@@ -0,0 +1,343 @@
+class ChainNetworkHookWithFFHookEventsTestCase(TestCase):
+    @responses.activate
+    def test_on_chain_create(self) -> None:
+        chain_id = fake.pyint()
+        responses.add(
+            responses.POST,
+            "http://127.0.0.1/v1/hooks/events",
+            status=200,
+            match=[
+                responses.matchers.header_matcher(
+                    {"Authorization": "Basic example-token"}
+                ),
+                responses.matchers.json_params_matcher(
+                    {"type": "CHAIN_UPDATE", "chain_id": str(chain_id)}
+                ),
+            ],
+        )
+
+        ChainFactory.create(id=chain_id)
+
+        assert len(responses.calls) == 1
+        assert isinstance(responses.calls[0], responses.Call)
+        assert responses.calls[
+            0
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain_id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[0].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[0].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+    @responses.activate
+    def test_on_chain_delete(self) -> None:
+        chain = ChainFactory.create()
+
+        chain.delete()
+
+        # 2 calls: one for creation and one for deletion
+        assert len(responses.calls) == 2
+
+    @responses.activate
+    def test_on_chain_update(self) -> None:
+        chain = ChainFactory.create()
+
+        # Not updating using queryset because hooks are not triggered that way
+        chain.currency_name = "Ether"
+        chain.save()
+
+        # 2 calls: one for creation and one for updating
+        assert len(responses.calls) == 2
+
+
+@override_settings(
+    CGW_URL="http://127.0.0.1", CGW_FLUSH_TOKEN="example-token", FF_HOOK_EVENTS=True
+)
+class FeatureHookTestCase(TestCase):
+    @responses.activate
+    def test_on_feature_create(self) -> None:
+        Feature(key="Test Feature").save()
+
+        # Creating a feature with no chains should not trigger any webhook
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_feature_create_with_chain(self) -> None:
+        chain = ChainFactory.create()
+        FeatureFactory.create(key="Test Feature", chains=(chain,))
+
+        # 1 call for Chain creation, 1 call for feature creation, 1 call for M2M update
+        assert len(responses.calls) == 3
+        assert isinstance(responses.calls[2], responses.Call)
+        assert responses.calls[
+            2
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain.id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[0].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[0].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+    @responses.activate
+    def test_on_feature_delete(self) -> None:
```

Comment:
Makes sense 👍 I'll update them.

Reply:
Changed naming here: [d465bf0](https://github.com/safe-global/safe-config-service/pull/964/commits/d465bf08bcf9c918fa2354a23c992fc9f775f40a)

Also removed the `hook_call` suffix because I think it's not required given the name os the tests.